src/components/projectSelect/index.tsx

```diff
@@ -66,7 +70,14 @@ export const ProjectSelect: React.FC<ProjectSelectProps> = ({
       options={allProjects}
       onChange={(projectIdentifier: any) => {
         Cookies.set(CURRENT_PROJECT, projectIdentifier, { expires: 365 });
-        history.push(getCommitsRoute(projectIdentifier));
+        history.push(
+          isProjectSettingsPage
+            ? getProjectSettingsRoute(
+                projectIdentifier,
+                ProjectSettingsTabRoutes.General
+              )
+            : getCommitsRoute(projectIdentifier)
+        );
```

Comment:
Instead of casing on it we should pass this callback in as a parameter. So the component could be used more widely.

Reply:
it's pretty heavily cased otherwise, so I don't know if this will do much. If this is needed elsewhere, it'll require more work anyhow.

Reply:
I think as a general rule of reusable component design it would make sense to make this more generic. This component will also be used on the project patches page eventually.

Reply:
done